PerfDiagnosticListener/PerfDiagnosticListener.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DiagnosticAdapter;
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+
+namespace PerfDiagnosticListener
+{
+    public class PerfDiagnosticListener
+    {
+        private static long _requestsBegan = 0;
+        private static long _requestsEnded = 0;
+        private static long _requestTicks = 0;
+
+        [DiagnosticName("Microsoft.AspNetCore.Hosting.BeginRequest")]
+        public void OnBeginRequest(HttpContext httpContext, long timestamp)
+        {
+            Interlocked.Increment(ref _requestsBegan);
+
+            var requestInfo = new RequestInfo { BeginTimestamp = timestamp };
+            httpContext.Features.Set(requestInfo);
```

Comment:
I will make this change, but then we should also change https://github.com/Microsoft/ApplicationInsights-aspnetcore/blob/develop/src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs in the same way.  Were we involved in the implementation of this?  @pakrym